####Big Data Analysis  
Drugi deo projekta u kome se zahteva definisanje i analiza hipoteza  
Bići obrađeni transformisani podaci o uticaju stresa na čoveka

Učitavanje transformisanih podataka u Notebook

In [0]:
from pyspark.sql import SparkSession
from pyspark.sql.functions import *
from pyspark.sql.types import *

In [0]:
all_data_df_path = 'dbfs:/stresTransformacije/df.csv'

all_data_df = spark.read.csv(all_data_df_path, header=True, inferSchema=True)

In [0]:
all_data_df.cache()
all_data_df.printSchema()
all_data_df.show()

root
 |-- hrkanje: double (nullable = true)
 |-- brzina_disanja: double (nullable = true)
 |-- temperatura_tela: double (nullable = true)
 |-- kiseonik_u_krvi: double (nullable = true)
 |-- dužina_sna: double (nullable = true)
 |-- puls: double (nullable = true)
 |-- stres: integer (nullable = true)
 |-- vlažnost: double (nullable = true)
 |-- broj_koraka: integer (nullable = true)
 |-- nivo_stresa: integer (nullable = true)

+-------+--------------+----------------+---------------+----------+-----+-----+--------+-----------+-----------+
|hrkanje|brzina_disanja|temperatura_tela|kiseonik_u_krvi|dužina_sna| puls|stres|vlažnost|broj_koraka|nivo_stresa|
+-------+--------------+----------------+---------------+----------+-----+-----+--------+-----------+-----------+
|   60.0|          20.0|            96.0|           95.0|       7.0| 60.0|    1|   15.97|         51|          1|
|  85.76|        23.536|          90.768|         88.768|     0.768|68.84|    3|   15.97|         51|          1|


####Vizualizacija data set-a

In [0]:
display(all_data_df)

hrkanje brzina_disanja temperatura_tela kiseonik_u_krvi dužina_sna puls stres vlažnost broj_koraka nivo_stresa 60.0 20.0 96.0 95.0 7.0 60.0 1 15.97 51 1 85.76 23.536 90.768 88.768 0.768 68.84 3 15.97 51 1 56.88 19.376 95.376 94.064 6.376 58.44 1 15.97 51 1 55.52 19.104 95.104 93.656 6.104 57.76 1 15.97 51 1 73.44 21.344 93.344 91.344 4.016 63.36 2 15.97 51 1 59.28 19.856 95.856 94.784 6.856 59.64 1 15.97 51 1 87.8 24.08 91.04 89.04 1.04 70.2 3 15.97 51 1 52.32 18.464 94.464 92.696 5.464 56.16 1 15.97 51 1 52.64 18.528 94.528 92.792 5.528 56.32 1 15.97 51 1 86.24 23.664 90.832 88.832 0.832 69.16 3 15.97 51 1 81.56 22.416 90.208 88.208 0.208 66.04 3 15.97 51 1 63.68 20.368 92.368 90.368 2.552 60.92 2 15.97 51 1 77.6 21.76 93.76 91.76 4.64 64.4 2 15.97 51 1 77.28 21.728 93.728 91.728 4.592 64.32 2 15.97 51 1 69.76 20.976 92.976 90.976 3.464 62.44 2 15.97 51 1 88.04 24.144 91.072 89.072 1.072 70.36 3 15.97 51 1 89.96 24.656 91.328 89.328 1.328 71.64 3 15.97 51 1 53.68 18.736 94.736 93.104 5.736 56.84 1 15.97 51 1 78.56 21.856 93.856 91.856 4.784 64.64 2 15.97 51 1 50.96 18.192 94.192 92.288 5.192 55.48 1 15.97 51 1 62.08 20.208 92.208 90.208 2.312 60.52 2 15.97 51 1 60.0 20.0 92.0 90.0 2.0 60.0 2 15.97 51 1 86.72 23.792 90.896 88.896 0.896 69.48 3 15.97 51 1 60.96 20.096 92.096 90.096 2.144 60.24 2 15.97 51 1 65.6 20.56 92.56 90.56 2.84 61.4 2 15.97 51 1 75.36 21.536 93.536 91.536 4.304 63.84 2 15.97 51 1 73.28 21.328 93.328 91.328 3.992 63.32 2 15.97 51 1 62.88 20.288 92.288 90.288 2.432 60.72 2 15.97 51 1 56.72 19.344 95.344 94.016 6.344 58.36 1 15.97 51 1 53.6 18.72 94.72 93.08 5.72 56.8 1 15.97 51 1 56.08 19.216 95.216 93.824 6.216 58.04 1 15.97 51 1 75.68 21.568 93.568 91.568 4.352 63.92 2 15.97 51 1 87.56 24.016 91.008 89.008 1.008 70.04 3 15.97 51 1 58.48 19.696 95.696 94.544 6.696 59.24 1 15.97 51 1 81.08 22.288 90.144 88.144 0.144 65.72 3 15.97 51 1 61.6 20.16 92.16 90.16 2.24 60.4 2 15.97 51 1 50.24 18.048 94.048 92.072 5.048 55.12 1 15.97 51 1 60.48 20.048 92.048 90.048 2.072 60.12 2 15.97 51 1 50.48 18.096 94.096 92.144 5.096 55.24 1 15.97 51 1 57.68 19.536 95.536 94.304 6.536 58.84 1 15.97 51 1 56.48 19.296 95.296 93.944 6.296 58.24 1 15.97 51 1 90.08 24.688 91.344 89.344 1.344 71.72 3 15.97 51 1 50.64 18.128 94.128 92.192 5.128 55.32 1 15.97 51 1 50.56 18.112 94.112 92.168 5.112 55.28 1 15.97 51 1 53.92 18.784 94.784 93.176 5.784 56.96 1 15.97 51 1 68.32 20.832 92.832 90.832 3.248 62.08 2 15.97 51 1 54.88 18.976 94.976 93.464 5.976 57.44 1 15.97 51 1 52.4 18.48 94.48 92.72 5.48 56.2 1 15.97 51 1 51.6 18.32 94.32 92.48 5.32 55.8 1 15.97 51 1 65.12 20.512 92.512 90.512 2.768 61.28 2 15.97 51 1 86.6 23.76 90.88 88.88 0.88 69.4 3 15.97 51 1 68.0 20.8 92.8 90.8 3.2 62.0 2 15.97 51 1 63.2 20.32 92.32 90.32 2.48 60.8 2 15.97 51 1 75.2 21.52 93.52 91.52 4.28 63.8 2 15.97 51 1 76.96 21.696 93.696 91.696 4.544 64.24 2 15.97 51 1 80.0 22.0 90.0 88.0 0.0 65.0 3 15.97 51 1 53.52 18.704 94.704 93.056 5.704 56.76 1 15.97 51 1 61.44 20.144 92.144 90.144 2.216 60.36 2 15.97 51 1 62.24 20.224 92.224 90.224 2.336 60.56 2 15.97 51 1 79.2 21.92 93.92 91.92 4.88 64.8 2 15.97 51 1 54.4 18.88 94.88 93.32 5.88 57.2 1 15.97 51 1 68.96 20.896 92.896 90.896 3.344 62.24 2 15.97 51 1 90.56 24.816 91.408 89.408 1.408 72.04 3 15.97 51 1 66.88 20.688 92.688 90.688 3.032 61.72 2 15.97 51 1 68.64 20.864 92.864 90.864 3.296 62.16 2 15.97 51 1 76.0 21.6 93.6 91.6 4.4 64.0 2 15.97 51 1 54.96 18.992 94.992 93.488 5.992 57.48 1 15.97 51 1 55.28 19.056 95.056 93.584 6.056 57.64 1 15.97 51 1 90.32 24.752 91.376 89.376 1.376 71.88 3 15.97 51 1 58.08 19.616 95.616 94.424 6.616 59.04 1 15.97 51 1 83.6 22.96 90.48 88.48 0.48 67.4 3 15.97 51 1 65.76 20.576 92.576 90.576 2.864 61.44 2 15.97 51 1 57.2 19.44 95.44 94.16 6.44 58.6 1 15.97 51 1 68.8 20.88 92.88 90.88 3.32 62.2 2 15.97 51 1 59.68 19.936 95.936 94.904 6.936 59.84 1 15.97 51 1 78.08 21.808 93.808 91.808 4.712 64.52 2 15.97 51 1 71.36 21.136 93.136 91.136 3.704 62.84 2 15.97 51 1 81.44 22.384 90.192 88.19

Broj osoba sa stresom grupisan po nivoima stresa

In [0]:
display(all_data_df.select('stres').groupBy('stres').count().orderBy('count', ascending=False))

stres count 2 162640 1 161350 3 121335

Output can only be rendered in Databricks

Broj koraka grupisan po nivou stresa

In [0]:
display(all_data_df.select('stres', 'broj_koraka').where(col('stres') == 2)
       .groupBy('stres', 'broj_koraka')
       .count().orderBy('count', ascending=False))

stres broj_koraka count 2 125 1386 2 127 1386 2 126 1386 2 124 1386 2 123 1386 2 128 1386 2 122 1386 2 183 1260 2 192 1260 2 107 1260 2 166 1260 2 170 1260 2 137 1260 2 101 1260 2 105 1260 2 162 1260 2 144 1260 2 194 1260 2 161 1260 2 110 1260 2 103 1260 2 158 1260 2 102 1260 2 159 1260 2 136 1260 2 152 1260 2 116 1260 2 117 1260 2 131 1260 2 150 1260 2 184 1260 2 151 1260 2 185 1260 2 145 1260 2 182 1260 2 191 1260 2 130 1260 2 111 1260 2 153 1260 2 198 1260 2 104 1260 2 113 1260 2 167 1260 2 134 1260 2 169 1260 2 186 1260 2 109 1260 2 138 1260 2 106 1260 2 118 1260 2 108 1260 2 193 1260 2 121 1260 2 119 1260 2 199 1260 2 200 1260 2 176 1260 2 168 1260 2 142 1260 2 135 1260 2 129 1260 2 100 1260 2 115 1260 2 174 1260 2 190 1260 2 120 1260 2 154 1260 2 178 1260 2 143 1260 2 177 1260 2 114 1260 2 146 1260 2 112 1260 2 160 1260 2 175 1260 2 187 1259 2 181 1259 2 165 1259 2 188 1259 2 148 1259 2 157 1259 2 141 1259 2 195 1259 2 163 1259 2 164 1259 2 132 1259 2 133 1259 2 149 1259 2 197 1259 2 155 1259 2 196 1259 2 171 1259 2 147 1259 2 156 1259 2 139 1259 2 173 1259 2 140 1259 2 189 1259 2 172 1259 2 179 1259 2 180 1259 2 94 1134 2 92 1134 2 98 1134 2 96 1134 2 93 1134 2 95 1134 2 97 1134 2 99 1134 2 80 630 2 77 630 2 56 630 2 78 630 2 67 630 2 57 630 2 74 630 2 63 630 2 64 630 2 85 630 2 51 630 2 69 630 2 90 630 2 66 630 2 58 630 2 59 630 2 52 630 2 86 630 2 55 630 2 84 630 2 88 630 2 76 630 2 65 630 2 70 630 2 60 630 2 73 630 2 75 630 2 87 630 2 68 630 2 62 630 2 72 630 2 81 630 2 82 630 2 71 630 2 83 630 2 53 630 2 61 630 2 79 630 2 54 630 2 89 630 2 91 252

Output can only be rendered in Databricks

Broj stresa grupisan u odnosu na dužinu spavanja

In [0]:
display(all_data_df.select('stres').where(col('dužina_sna') < 4)
       .groupBy('stres')
       .count().orderBy('count', ascending=False))

stres count 3 121335 2 108427

Output can only be rendered in Databricks

Količina stresa u momentu kada se ubrza disanje

In [0]:
display(all_data_df.select('stres', 'brzina_disanja')
       .where(col('brzina_disanja') > 0)
       .groupBy('stres').count().orderBy('count'), ascending=False)

stres count 3 121335 1 161350 2 162640

Output can only be rendered in Databricks

Količina vlažnosti grupisana po nivou stresa

In [0]:
display(all_data_df.select('stres', 'vlažnost')
       .where(col('vlažnost') > 20)
       .groupBy('stres').count().orderBy('count'), ascending=False)

stres count 3 93981 1 124975 2 125974

Output can only be rendered in Databricks

Uticaj glasnoće hrkanja na dužinu spavanja

In [0]:

display(all_data_df.groupBy("dužina_sna").agg({"hrkanje": "avg"}))

dužina_sna avg(hrkanje) 4.592 77.28000000000011 5.48 52.3999999999998 0.032 80.24000000000004 6.936 59.68000000000017 5.608 53.04000000000026 4.88 79.2000000000003 6.44 57.20000000000018 0.0 80.0 3.368 69.12000000000032 3.776 71.84000000000012 5.4 52.0 0.576 84.3199999999998 7.0 60.0 0.16 81.20000000000032 5.848 54.23999999999983 3.08 67.20000000000024 5.752 53.76000000000017 1.168 88.75999999999982 2.288 61.92000000000006 6.008 55.04000000000027 0.352 82.63999999999976 3.56 70.39999999999978 2.024 60.15999999999988 4.376 75.84000000000012 1.104 88.27999999999997 0.688 85.15999999999988 0.416 83.12000000000012 2.84 65.60000000000024 0.144 81.08000000000018 6.152 55.76000000000017 4.448 76.31999999999981 6.6 58.0 1.44 90.79999999999963 6.472 57.36000000000012 4.328 75.52000000000032 4.088 73.91999999999994 2.792 65.28000000000023 6.376 56.879999999999775 5.88 54.3999999999998 5.624 53.12000000000019 5.224 51.12000000000017 6.28 56.399999999999785 5.208 51.040000000000234 4.256 75.04000000000042 1.184 88.87999999999994 5.672 53.3600000000001 1.424 90.6800000000002 2.168 61.12000000000026 6.888 59.44000000000008 5.032 50.15999999999988 5.56 52.79999999999986 6.696 58.47999999999971 5.048 50.239999999999824 2.648 64.31999999999982 0.72 85.40000000000006 0.672 85.04000000000048 4.496 76.63999999999977 4.832 78.87999999999984 5.96 54.79999999999984 6.344 56.71999999999982 2.336 62.23999999999982 0.624 84.6800000000002 0.24 81.79999999999967 2.96 66.39999999999978 0.128 80.95999999999955 3.656 71.04000000000038 5.784 53.920000000000066 1.216 89.12000000000003 6.104 55.52000000000027 1.376 90.3199999999998 6.744 58.719999999999814 6.072 55.360000000000106 5.656 53.280000000000165 0.256 81.9199999999998 2.552 63.680000000000184 5.576 52.87999999999981 3.608 70.71999999999977 4.136 74.23999999999994 2.36 62.399999999999764 5.32 51.6000000000002 0.208 81.56000000000007 4.544 76.9599999999996 2.192 61.280000000000214 2.576 63.84000000000012 5.176 50.87999999999983 4.112 74.08000000000006 3.632 70.87999999999973 5.24 51.20000000000011 6.92 59.60000000000023 2.264 61.76000000000018 5.928 54.639999999999894 3.992 73.28000000000017 6.856 59.280000000000214 5.592 52.95999999999974 2.528 63.5200000000003 6.424 57.120000000000225 4.712 78.08000000000014 6.408 57.04000000000029 0.64 84.79999999999967 3.128 67.52000000000031 0.8 86.0 2.888 65.92000000000006 0.752 85.63999999999974 0.064 80.47999999999968 5.288 51.44000000000004 6.136 55.68000000000016 6.664 58.31999999999983 5.944 54.719999999999835 4.928 79.52000000000032 0.432 83.2400000000001 4.52 76.7999999999997 0.512 83.84000000000012 4.304 75.36000000000023 0.464 83.47999999999966 6.616 58.07999999999994 5.256 51.28000000000014 5.688 53.440000000000055 5.08 50.3999999999998 5.112 50.559999999999974 2.768 65.12000000000029 2.456 63.04000000000035 3.104 67.36000000000017 1.488 91.15999999999988 5.816 54.079999999999934 0.816 86.12000000000008 0.384 82.87999999999988 4.352 75.68000000000018 0.304 82.28000000000006 3.872 72.47999999999968 0.88 86.59999999999994 3.512 70.07999999999997 2.672 64.47999999999969 5.448 52.23999999999983 0.112 80.84000000000012 4.208 74.71999999999983 0.608 84.5600000000001 4.784 78.56000000000002 3.032 66.8799999999997 4.616 77.44 6.568 57.84000000000012 5.096 50.47999999999973 4.976 79.84000000000012 2.12 60.799999999999784 6.312 56.559999999999924 1.456 90.91999999999967 5.896 54.47999999999973 6.296 56.47999999999971 6.456 57.280000000000186 6.2 56.0 6.824 59.120000000000246 6.216 56.07999999999994 1.408 90.56000000000017 3.296 68.63999999999982 4.688 77.91999999999987 4.856 79.04000000000043 3.944 72.95999999999961 5.416 52.079999999999934 3.8 72.0 5.992 54.95999999999973 0.896 86.72000000000001 1.136 88.52000000000034 0.96 87.20000000000036 5.64 53.200000000000145 0.992 87.43999999999986 1.312 89.84000000000012 6.024 55.12000000000021 6.776 58.87999999999977 3.056 67.04000000000036 6.168 55.84000000000012 5.864 54.31999999999984 3.176 67.84000000000012 4.568 77

Output can only be rendered in Databricks

Količina kiseonika grupisana po pulsu

In [0]:
display(all_data_df.select('kiseonik_u_krvi', 'puls')
       .where(col('puls') > 50)
       .groupBy('kiseonik_u_krvi').count().orderBy('count'), ascending=False)

kiseonik_u_krvi count 93.56 1290 91.92 1290 90.32 1290 89.36 1290 88.56 1290 88.0 1290 88.32 1290 91.68 1290 88.8 1290 92.36 1290 89.2 1290 90.16 1290 90.08 1290 88.72 1290 94.4 1290 90.8 1290 92.24 1290 93.68 1290 92.72 1290 91.36 1290 93.2 1290 94.76 1290 93.32 1290 91.52 1290 93.08 1290 94.64 1290 88.64 1290 88.88 1290 92.84 1290 91.04 1290 90.4 1290 88.48 1290 90.64 1290 90.48 1290 91.12 1290 93.92 1290 89.28 1290 88.96 1290 94.52 1290 92.96 1290 90.96 1290 94.04 1290 88.24 1290 94.88 1290 92.6 1290 90.88 1290 91.44 1290 94.28 1290 93.44 1290 89.04 1290 89.12 1290 90.72 1290 92.12 1290 91.6 1290 88.16 1290 91.76 1290 95.0 1290 90.24 1290 92.48 1290 92.0 1290 94.16 1290 91.84 1290 93.8 1290 88.4 1290 91.2 1290 88.08 1290 90.56 1290 91.28 1290 89.44 1290 90.0 1290 91.904 1291 89.488 1291 90.336 1291 92.024 1291 90.864 1291 88.848 1291 91.888 1291 89.328 1291 90.112 1291 94.976 1291 92.384 1291 88.688 1291 88.112 1291 90.832 1291 93.176 1291 91.216 1291 90.752 1291 89.344 1291 88.528 1291 88.768 1291 94.784 1291 90.352 1291 92.336 1291 90.016 1291 90.672 1291 88.304 1291 89.392 1291 94.808 1291 88.464 1291 88.032 1291 91.776 1291 91.424 1291 91.552 1291 90.304 1291 94.304 1291 91.008 1291 93.416 1291 94.592 1291 90.512 1291 91.952 1291 92.096 1291 88.944 1291 88.416 1291 89.104 1291 91.712 1291 88.496 1291 90.288 1291 93.008 1291 90.272 1291 90.176 1291 90.032 1291 92.048 1291 94.544 1291 91.136 1291 88.992 1291 93.704 1291 90.576 1291 88.336 1291 90.256 1291 88.976 1291 89.072 1291 93.488 1291 91.296 1291 89.008 1291 90.608 1291 94.664 1291 90.384 1291 94.736 1291 92.744 1291 91.088 1291 94.952 1291 90.128 1291 94.136 1291 90.544 1291 88.128 1291 90.768 1291 93.896 1291 91.344 1291 91.792 1291 92.936 1291 94.352 1291 94.424 1291 88.192 1291 88.752 1291 93.824 1291 91.936 1291 88.256 1291 90.816 1291 90.224 1291 90.992 1291 92.144 1291 91.808 1291 94.256 1291 94.088 1291 90.704 1291 91.824 1291 88.384 1291 90.464 1291 91.632 1291 92.864 1291 90.416 1291 89.248 1291 93.536 1291 90.064 1291 94.376 1291 88.272 1291 94.448 1291 89.376 1291 88.016 1291 94.688 1291 90.784 1291 93.296 1291 89.136 1291 93.848 1291 94.568 1291 94.616 1291 88.144 1291 92.168 1291 94.184 1291 91.072 1291 90.624 1291 93.776 1291 93.512 1291 91.856 1291 92.648 1291 88.512 1291 91.104 1291 90.848 1291 91.248 1291 91.264 1291 90.208 1291 90.912 1291 88.672 1291 94.904 1291 91.376 1291 89.424 1291 89.184 1291 91.456 1291 92.888 1291 90.976 1291 93.272 1291 89.472 1291 93.608 1291 88.576 1291 90.688 1291 91.504 1291 89.056 1291 90.448 1291 91.408 1291 88.096 1291 88.176 1291 88.368 1291 91.744 1291 94.832 1291 90.592 1291 93.656 1291 93.992 1291 92.504 1291 92.816 1291 91.152 1291 88.448 1291 90.496 1291 89.312 1291 93.224 1291 88.704 1291 90.736 1291 93.968 1291 92.216 1291 90.144 1291 92.312 1291 91.056 1291 93.728 1291 89.168 1291 88.048 1291 91.984 1291 92.456 1291 91.392 1291 92.192 1291 92.576 1291 91.584 1291 94.928 1291 91.568 1291 90.656 1291 92.528 1291 88.816 1291 88.624 1291 90.432 1291 90.048 1291 94.496 1291 93.872 1291 94.472 1291 88.928 1291 93.464 1291 93.368 1291 93.752 1291 89.408 1291 92.624 1291 90.192 1291 92.072 1291 90.928 1291 90.896 1291 94.232 1291 92.792 1291 92.288 1291 92.432 1291 91.648 1291 90.368 1291 93.944 1291 90.528 1291 91.472 1291 88.544 1291 92.696 1291 92.552 1291 91.536 1291 93.056 1291 93.032 1291 93.584 1291 92.984 1291 92.768 1291 88.592 1291 88.784 1291 91.024 1291 94.208 1291 92.264 1291 93.104 1291 94.112 1291 88.832 1291 91.968 1291 89.264 1291 88.912 1291 88.208 1291 94.016 1291 91.168 1291 94.856 1291 88.896 1291 89.152 1291 88.432 1291 88.656 1291 93.128 1291 90.096 1291 91.872 1291 91.616 1291 93.632 1291 91.232 1291 91.328 1291 94.064 1291 89.456 1291 91.312 1291 92.912 1291 91.664 1291 88.288 1291 88.608 1291 91.488 1291 88.352 1291 93.392 1291 89.232 1291 91.728 1291 90.944 1291 93.344 1291 89.296 1291 88.736 1291 93.248 1291 94.328 1291 93.152 1291 91.184 1291 94.712 1291 89.088 1291 88.864 1291 89.024 

Output can only be rendered in Databricks

Temperatura tela grupisana po vlažnosti

In [0]:
display(all_data_df.select('temperatura_tela')
        .where(col('vlažnost') > 0)
        .groupBy('temperatura_tela')
        .count()
        .orderBy('count', ascending=False))

temperatura_tela count 94.704 1291 90.336 1291 93.424 1291 93.216 1291 90.864 1291 95.856 1291 93.792 1291 92.256 1291 95.808 1291 95.328 1291 94.848 1291 94.288 1291 94.976 1291 90.112 1291 92.384 1291 95.376 1291 95.696 1291 93.376 1291 95.152 1291 90.832 1291 91.216 1291 90.752 1291 92.928 1291 94.784 1291 90.352 1291 92.336 1291 94.528 1291 90.016 1291 90.672 1291 94.128 1291 91.424 1291 90.304 1291 94.576 1291 91.008 1291 94.512 1291 94.304 1291 94.592 1291 90.512 1291 92.304 1291 94.176 1291 92.416 1291 92.096 1291 93.456 1291 93.088 1291 93.008 1291 90.288 1291 90.272 1291 92.048 1291 90.176 1291 95.744 1291 90.032 1291 91.136 1291 94.544 1291 95.184 1291 90.576 1291 94.096 1291 90.256 1291 94.368 1291 93.488 1291 91.296 1291 93.664 1291 90.608 1291 90.384 1291 94.736 1291 92.608 1291 90.128 1291 91.088 1291 92.512 1291 90.544 1291 90.768 1291 95.904 1291 93.264 1291 95.248 1291 94.352 1291 91.344 1291 94.608 1291 93.824 1291 93.136 1291 95.968 1291 95.392 1291 95.584 1291 90.816 1291 90.224 1291 90.992 1291 92.144 1291 94.768 1291 92.544 1291 94.256 1291 94.272 1291 93.232 1291 90.704 1291 95.344 1291 93.696 1291 95.936 1291 90.464 1291 92.992 1291 92.864 1291 94.048 1291 94.336 1291 90.416 1291 93.856 1291 92.176 1291 92.272 1291 93.536 1291 90.064 1291 92.464 1291 94.448 1291 93.888 1291 94.688 1291 92.704 1291 95.488 1291 94.224 1291 90.784 1291 93.296 1291 91.072 1291 90.624 1291 93.552 1291 95.056 1291 93.776 1291 94.992 1291 95.024 1291 93.808 1291 95.264 1291 95.216 1291 91.104 1291 93.984 1291 94.656 1291 95.616 1291 91.264 1291 91.248 1291 90.848 1291 93.712 1291 90.208 1291 95.088 1291 90.912 1291 91.376 1291 92.944 1291 91.456 1291 90.976 1291 95.664 1291 92.784 1291 95.296 1291 90.448 1291 90.688 1291 91.408 1291 94.864 1291 94.896 1291 94.832 1291 90.592 1291 92.656 1291 92.816 1291 91.152 1291 92.128 1291 90.496 1291 93.408 1291 92.976 1291 93.936 1291 93.504 1291 93.072 1291 92.032 1291 90.144 1291 92.064 1291 93.968 1291 90.736 1291 93.728 1291 95.504 1291 93.952 1291 91.056 1291 94.192 1291 91.392 1291 95.536 1291 92.576 1291 92.192 1291 92.848 1291 94.928 1291 95.984 1291 92.896 1291 92.528 1291 90.656 1291 95.456 1291 92.224 1291 92.368 1291 95.792 1291 90.432 1291 94.496 1291 90.048 1291 93.872 1291 92.208 1291 93.168 1291 95.552 1291 90.192 1291 92.624 1291 94.944 1291 90.928 1291 90.896 1291 92.832 1291 95.072 1291 92.288 1291 92.432 1291 95.952 1291 94.672 1291 94.144 1291 93.024 1291 90.368 1291 94.384 1291 93.184 1291 90.528 1291 91.472 1291 93.056 1291 95.648 1291 93.584 1291 92.768 1291 94.624 1291 95.888 1291 95.824 1291 91.024 1291 94.208 1291 93.312 1291 93.104 1291 94.112 1291 93.472 1291 95.472 1291 92.736 1291 91.168 1291 92.592 1291 94.016 1291 95.104 1291 94.432 1291 95.408 1291 95.168 1291 95.008 1291 93.328 1291 90.096 1291 93.632 1291 91.232 1291 91.328 1291 92.688 1291 94.064 1291 94.912 1291 92.912 1291 91.312 1291 92.112 1291 94.816 1291 95.232 1291 94.752 1291 91.488 1291 93.392 1291 95.312 1291 93.616 1291 95.728 1291 93.344 1291 95.776 1291 90.944 1291 92.496 1291 94.464 1291 93.568 1291 93.248 1291 92.352 1291 93.904 1291 93.152 1291 95.632 1291 91.184 1291 95.872 1291 92.016 1291 95.568 1291 93.648 1291 94.032 1291 93.744 1291 95.712 1291 95.424 1291 95.136 1291 92.752 1291 94.416 1291 92.448 1291 92.672 1291 94.72 1290 93.36 1290 95.36 1290 92.88 1290 92.4 1290 93.84 1290 90.32 1290 92.16 1290 94.08 1290 95.52 1290 92.32 1290 94.96 1290 92.8 1290 90.16 1290 96.0 1290 90.08 1290 95.76 1290 93.12 1290 95.84 1290 95.92 1290 94.4 1290 90.8 1290 93.04 1290 92.24 1290 94.0 1290 93.68 1290 91.36 1290 92.72 1290 93.2 1290 95.6 1290 94.64 1290 94.56 1290 93.52 1290 91.04 1290 94.24 1290 90.4 1290 90.64 1290 95.28 1290 95.2 1290 90.48 1290 91.12 1290 93.92 1290 92.96 1290 92.56 1290 90.96 1290 92.64 1290 94.88 1290 90.88 1290 95.12 1290 91.44 1290 95.44 1290 93.44 1290 90.72 1290 94.8 1290 90.24 1290 95.04 1290 93.76 1290 92.0 1290 92.48 1290 94.16 1290 93.6 1290 91.2 1290 92.08 129

Output can only be rendered in Databricks

Broj koraka grupisan po vlažnosti

In [0]:
display(all_data_df.select('broj_koraka', 'vlažnost')
        .where(col('vlažnost') > 20)
        .groupBy('broj_koraka')
        .count()
        .orderBy('count', ascending=False))

broj_koraka count 137 3450 193 3450 192 3450 185 3450 169 3450 161 3450 162 3450 130 3450 194 3450 145 3450 200 3450 170 3450 131 3450 186 3450 153 3450 138 3450 177 3450 146 3450 178 3450 154 3450 148 3449 133 3449 155 3449 183 3449 159 3449 157 3449 140 3449 132 3449 152 3449 182 3449 168 3449 142 3449 164 3449 139 3449 191 3449 163 3449 165 3449 189 3449 179 3449 197 3449 196 3449 173 3449 176 3449 136 3449 171 3449 166 3449 181 3449 167 3449 156 3449 195 3449 141 3449 151 3449 188 3449 150 3449 147 3449 198 3449 199 3449 158 3449 135 3449 149 3449 184 3449 172 3449 144 3449 134 3449 187 3449 190 3449 175 3449 160 3449 143 3449 174 3449 180 3449 127 3105 125 3105 123 3105 126 3104 128 3104 124 3104 122 3104 111 2760 117 2760 107 2760 129 2760 113 2760 121 2760 109 2760 105 2760 119 2760 115 2760 110 2759 106 2759 116 2759 118 2759 108 2759 120 2759 112 2759 114 2759 101 2415 103 2415 100 2414 102 2414 104 2414 93 2070 95 2070 99 2070 97 2070 94 2069 96 2069 92 2069 98 2069

Output can only be rendered in Databricks

Količina kiseonika u krvi grupisana po stresu

In [0]:
display(all_data_df.select('stres', 'kiseonik_u_krvi')
        .where(col('kiseonik_u_krvi') > 0)
        .groupBy('stres')
        .count()
        .orderBy('count', ascending=False))

stres count 2 162640 1 161350 3 121335

Output can only be rendered in Databricks

##Analiza hipoteza

####Hipoteza 1  
Odnos stresa na dužinu sna.

In [0]:
display(all_data_df.select('stres')
       .where(col('dužina_sna') < 7)
       .groupBy('stres')
       .count()
       .orderBy('count', ascending=False))

stres count 2 162640 1 160060 3 121335

Output can only be rendered in Databricks

Pored dužine sna potrebno je da vidimo kako stres utiče na puls i nivo kiseonika u krvi.

In [0]:
broj_stresiranih = (all_data_df.select('stres')
    .where(col('stres') > 0)
    .count())
print('Ukupan broj osoba sa stresom: ', broj_stresiranih)

broj_pulsa_kiseonika = (all_data_df.select('stres')
                 .where(col('puls') > 0)
                 .where(col('kiseonik_u_krvi') > 0)
                 .where(col('stres') == 3)
                 .count())

print('Ukupan broj osoba sa određenim pulsom i kiseonikom kod kojih je stres visok: ', broj_pulsa_kiseonika)

Ukupan broj osoba sa stresom:  445325
Ukupan broj osoba sa određenim pulsom i kiseonikom kod kojih je stres visok:  121335


In [0]:
display(all_data_df.select('dužina_sna')
       .where(col('stres') == 3)
       .groupBy('dužina_sna')
       .count()
       .orderBy('count', ascending=False))

print('Prosek osoba kod kojih stres utiče na dužinu spavanja: ', broj_pulsa_kiseonika/broj_stresiranih*100, '%')

dužina_sna count 0.032 1291 0.576 1291 1.168 1291 0.352 1291 1.104 1291 0.688 1291 0.416 1291 0.144 1291 1.184 1291 1.424 1291 0.672 1291 0.624 1291 0.128 1291 1.216 1291 1.376 1291 0.256 1291 0.208 1291 0.752 1291 0.064 1291 0.432 1291 0.512 1291 0.464 1291 1.488 1291 0.816 1291 0.384 1291 0.304 1291 0.112 1291 0.608 1291 1.456 1291 1.408 1291 0.896 1291 1.136 1291 0.992 1291 1.312 1291 0.704 1291 0.496 1291 0.096 1291 0.048 1291 1.072 1291 1.344 1291 0.768 1291 1.152 1291 0.736 1291 1.328 1291 1.008 1291 1.248 1291 0.976 1291 1.392 1291 0.544 1291 1.232 1291 0.448 1291 0.912 1291 0.288 1291 0.224 1291 1.056 1291 1.024 1291 1.088 1291 0.848 1291 0.192 1291 0.784 1291 1.296 1291 0.832 1291 0.272 1291 0.592 1291 1.472 1291 0.864 1291 0.336 1291 0.368 1291 0.016 1291 0.176 1291 0.528 1291 0.656 1291 0.928 1291 1.264 1291 0.944 1291 0.0 1290 0.16 1290 1.44 1290 0.72 1290 0.24 1290 0.64 1290 0.8 1290 0.88 1290 0.96 1290 0.32 1290 0.4 1290 0.08 1290 0.56 1290 1.04 1290 1.36 1290 1.12 1290 1.2 1290 0.48 1290 1.28 1290

Output can only be rendered in Databricks

Prosek osoba kod kojih stres utiče na dužinu spavanja:  27.246393083702912 %


Prethodno pokazanom analizom dolazim do zaključka da od ukupno 445325 osoba koje imaju problema sa stresom njih 121335 ima vrlo visok novi stresa što pored spavanja dodatno utiče na povećanje pulsa kao i kiseonika u krvi. U procentima to iznosi oko 27%.  
Dakle ovim bismo mogli reći da stres utiče na dužinu sna i samim tim je hipoteza <b>tačna</b>.

####Hipoteza 2  
Nivo stresa je visok što je veća aktivnost.

In [0]:
display(all_data_df.select('stres')
       .where(col('broj_koraka') > 0)
       .groupBy('stres')
       .count()
       .orderBy('count', ascending=False))



stres count 2 162640 1 161350 3 121335

Output can only be rendered in Databricks

Proverićemo još da li vlažnost utiče na aktivnost i brzinu disanja kod osoba sa stresom.  
Napravio bih još dva data frame-a. Jedan data frame bi se odnosio na broj aktivnosti kada su uključeni parametri vlažnosti. Dok drugi data frame bi pokazao broj aktivnosti prouzrokovan brzinom disanja.

In [0]:
broj_aktivnosti_vl = (all_data_df.select('stres', 'broj_koraka')
                     .where(col('stres') == 3)
                     .where(col('vlažnost') > 25))

broj_aktivnosti_vl = (broj_aktivnosti_vl.groupBy('broj_koraka')).count()
broj_aktivnosti_vl = (broj_aktivnosti_vl.orderBy('broj_koraka')
                      .withColumnRenamed('count', 'broj_aktivnosti_vl'))

display(broj_aktivnosti_vl)

broj_koraka broj_aktivnosti_vl 130 658 131 658 132 658 133 658 134 658 135 658 136 658 137 658 138 658 139 658 140 658 141 658 142 658 143 658 144 658 145 658 146 658 147 658 148 658 149 658 150 658 151 658 152 658 153 658 154 658 155 658 156 658 157 658 158 658 159 658 160 658 161 658 162 658 163 658 164 658 165 658 166 658 167 658 168 658 169 658 170 658 171 658 172 658 173 658 174 658 175 658 176 658 177 658 178 658 179 658 180 658 181 658 182 658 183 658 184 658 185 658 186 658 187 658 188 658 189 658 190 658 191 658 192 658 193 658 194 658 195 658 196 658 197 658 198 752 199 752 200 752

Output can only be rendered in Databricks

In [0]:
broj_aktivnosti_bd = (all_data_df.select('stres', 'broj_koraka')
                     .where(col('stres') == 3)
                     .where(col('brzina_disanja') > 0))

broj_aktivnosti_bd = (broj_aktivnosti_bd.groupBy('broj_koraka')).count()
broj_aktivnosti_bd = (broj_aktivnosti_bd.orderBy('broj_koraka')
                      .withColumnRenamed('count', 'broj_aktivnosti_bd'))

display(broj_aktivnosti_bd)

broj_koraka broj_aktivnosti_bd 51 470 52 470 53 470 54 470 55 470 56 470 57 470 58 470 59 470 60 470 61 470 62 470 63 470 64 470 65 470 66 470 67 470 68 470 69 470 70 470 71 470 72 470 73 470 74 470 75 470 76 470 77 470 78 470 79 470 80 470 81 470 82 470 83 470 84 470 85 470 86 470 87 470 88 470 89 470 90 470 91 188 92 845 93 846 94 845 95 846 96 845 97 846 98 845 99 846 100 939 101 940 102 939 103 940 104 939 105 940 106 939 107 940 108 939 109 940 110 939 111 940 112 939 113 940 114 939 115 940 116 939 117 940 118 939 119 940 120 939 121 940 122 1033 123 1034 124 1033 125 1034 126 1033 127 1034 128 1033 129 940 130 940 131 940 132 940 133 940 134 940 135 940 136 940 137 940 138 940 139 940 140 940 141 940 142 940 143 940 144 940 145 940 146 940 147 940 148 940 149 940 150 940 151 940 152 940 153 940 154 940 155 940 156 940 157 940 158 940 159 940 160 940 161 940 162 940 163 940 164 940 165 940 166 940 167 940 168 940 169 940 170 940 171 940 172 940 173 940 174 940 175 940 176 940 177 940 178 940 179 940 180 940 181 940 182 940 183 940 184 940 185 940 186 940 187 940 188 940 189 940 190 940 191 940 192 940 193 940 194 940 195 940 196 940 197 940 198 940 199 940 200 940

Output can only be rendered in Databricks

In [0]:
all_data_df.select(avg('broj_koraka').alias('prosek')).show()
broj_aktivnosti_vl.select(avg('broj_koraka').alias('prosek aktivnosti - vlažnost')).show()
broj_aktivnosti_bd.select(avg('broj_koraka').alias('prosek aktivnosti - disanje')).show()

+------------------+
|            prosek|
+------------------+
|134.41504743726492|
+------------------+

+----------------------------+
|prosek aktivnosti - vlažnost|
+----------------------------+
|                       165.0|
+----------------------------+

+---------------------------+
|prosek aktivnosti - disanje|
+---------------------------+
|                      125.5|
+---------------------------+



Dolazimo do zaključka da prvobitna prosečna aktivnost iznosi 134 koraka uzimajući u obzir celokupan data set. U što precizniju analizu podataka uključujemo parametre vlažnost i brzinu disanja, kojima bismo pokazali i još bolje utvrdili da li će zajedno sa njima naš član u hipotezi srazmerno rasti.  
Time smo zaključili da veća vlažnost nam pokazuje veći broj koraka pri visokom stresu, ali još jedna pretpostavka u analizi koju smo koristili nam daje rezultat pri kojem dobijamo znatno manji broj koraka od prosečnog.   
Na kraju možemo konstatovati da je hipoteza <b>netačna</b>

####Hipoteza 3  
Visok stres prouzrokuje povećanje temperature i smanjenje kiseonika u krvi.

Prikazaćemo broj osoba sa stresom u odnosu na temperaturu, kao i u odnosu na kiseonik u krvi.

In [0]:
broj_temp_df = (all_data_df.select('stres', 'temperatura_tela')
               .where(col('stres') > 0))
broj_temp_df = (broj_temp_df.groupBy(col('temperatura_tela')
                                     .alias('temperatura_tela')).count())

print('Broj osoba koda koje imaju stres: ', broj_temp_df.count())
display(broj_temp_df)

Broj osoba koda koje imaju stres:  345


temperatura_tela count 94.704 1291 90.336 1291 93.424 1291 93.216 1291 90.864 1291 95.856 1291 93.792 1291 94.72 1290 92.256 1291 95.808 1291 95.328 1291 94.848 1291 94.288 1291 94.976 1291 90.112 1291 92.384 1291 95.376 1291 95.696 1291 93.376 1291 95.152 1291 90.832 1291 93.36 1290 95.36 1290 92.88 1290 91.216 1291 90.752 1291 92.928 1291 94.784 1291 90.352 1291 92.4 1290 92.336 1291 94.528 1291 93.84 1290 90.016 1291 90.672 1291 90.32 1290 92.16 1290 94.128 1291 91.424 1291 90.304 1291 94.576 1291 91.008 1291 94.512 1291 94.08 1290 94.304 1291 94.592 1291 90.512 1291 92.304 1291 94.176 1291 92.416 1291 92.096 1291 93.456 1291 95.52 1290 93.088 1291 92.32 1290 93.008 1291 90.288 1291 90.272 1291 92.048 1291 90.176 1291 95.744 1291 90.032 1291 91.136 1291 94.96 1290 94.544 1291 95.184 1291 90.576 1291 94.096 1291 92.8 1290 90.16 1290 90.256 1291 94.368 1291 93.488 1291 91.296 1291 93.664 1291 90.608 1291 90.384 1291 96.0 1290 90.08 1290 94.736 1291 92.608 1291 90.128 1291 91.088 1291 95.76 1290 92.512 1291 90.544 1291 90.768 1291 95.904 1291 93.264 1291 95.248 1291 94.352 1291 91.344 1291 93.12 1290 95.84 1290 95.92 1290 94.608 1291 93.824 1291 93.136 1291 95.968 1291 95.392 1291 94.4 1290 95.584 1291 90.816 1291 90.224 1291 90.992 1291 90.8 1290 93.04 1290 92.144 1291 94.768 1291 92.24 1290 92.544 1291 94.256 1291 94.272 1291 93.232 1291 94.0 1290 90.704 1291 95.344 1291 93.68 1290 93.696 1291 95.936 1291 90.464 1291 91.36 1290 92.992 1291 92.72 1290 92.864 1291 93.2 1290 94.048 1291 94.336 1291 90.416 1291 93.856 1291 92.176 1291 92.272 1291 93.536 1291 90.064 1291 95.6 1290 92.464 1291 94.448 1291 93.888 1291 94.688 1291 92.704 1291 95.488 1291 94.224 1291 90.784 1291 93.296 1291 94.64 1290 91.072 1291 94.56 1290 90.624 1291 93.552 1291 95.056 1291 93.776 1291 94.992 1291 95.024 1291 93.52 1290 93.808 1291 95.264 1291 95.216 1291 91.104 1291 93.984 1291 94.656 1291 95.616 1291 91.264 1291 91.248 1291 90.848 1291 93.712 1291 90.208 1291 95.088 1291 90.912 1291 91.376 1291 92.944 1291 91.456 1291 91.04 1290 90.976 1291 95.664 1291 92.784 1291 94.24 1290 95.296 1291 90.448 1291 90.688 1291 90.4 1290 91.408 1291 94.864 1291 90.64 1290 94.896 1291 94.832 1291 90.592 1291 95.28 1290 95.2 1290 90.48 1290 91.12 1290 92.656 1291 92.816 1291 91.152 1291 92.128 1291 90.496 1291 93.408 1291 92.976 1291 93.936 1291 93.504 1291 93.072 1291 92.032 1291 90.144 1291 93.92 1290 92.064 1291 93.968 1291 90.736 1291 93.728 1291 95.504 1291 93.952 1291 91.056 1291 94.192 1291 91.392 1291 95.536 1291 92.576 1291 92.192 1291 92.848 1291 94.928 1291 95.984 1291 92.96 1290 92.896 1291 92.528 1291 90.656 1291 92.56 1290 95.456 1291 90.96 1290 92.224 1291 92.64 1290 92.368 1291 95.792 1291 90.432 1291 94.88 1290 94.496 1291 90.048 1291 93.872 1291 92.208 1291 93.168 1291 90.88 1290 95.552 1291 90.192 1291 92.624 1291 94.944 1291 95.12 1290 90.928 1291 91.44 1290 90.896 1291 92.832 1291 95.44 1290 95.072 1291 92.288 1291 92.432 1291 95.952 1291 94.672 1291 93.44 1290 94.144 1291 93.024 1291 90.368 1291 94.384 1291 93.184 1291 90.528 1291 91.472 1291 93.056 1291 95.648 1291 90.72 1290 93.584 1291 92.768 1291 94.624 1291 95.888 1291 95.824 1291 91.024 1291 94.8 1290 94.208 1291 93.312 1291 93.104 1291 94.112 1291 93.472 1291 90.24 1290 95.472 1291 92.736 1291 91.168 1291 92.592 1291 94.016 1291 95.104 1291 95.04 1290 93.76 1290 92.0 1290 94.432 1291 92.48 1290 95.408 1291 95.168 1291 95.008 1291 94.16 1290 93.328 1291 93.6 1290 90.096 1291 93.632 1291 91.2 1290 91.232 1291 91.328 1291 92.688 1291 94.064 1291 94.912 1291 92.08 1290 92.912 1291 91.312 1291 90.56 1290 92.112 1291 95.68 1290 94.816 1291 94.32 1290 95.232 1291 94.752 1291 91.28 1290 91.488 1291 93.392 1291 95.312 1291 93.616 1291 95.728 1291 93.344 1291 95.776 1291 90.944 1291 92.496 1291 94.464 1291 93.568 1291 94.48 1290 93.248 1291 92.352 1291 93.904 1291 93.152 1291 95.632 1291 93.28 1290 91.184 1291 95.872 1291 92.016 1291 95.568 1291 93.648 1291 94.032 1291 93.744 1291 95.712 1291 95.4

Output can only be rendered in Databricks

In [0]:
broj_kis_df = (all_data_df.select('stres', 'kiseonik_u_krvi')
               .where(col('stres') > 0))
broj_kis_df = (broj_kis_df.groupBy(col('kiseonik_u_krvi')
                                     .alias('kiseonik_u_krvi')).count())

print('Broj osoba koda koje imaju stres: ', broj_kis_df.count())
display(broj_kis_df)

Broj osoba koda koje imaju stres:  345


kiseonik_u_krvi count 91.904 1291 89.488 1291 90.336 1291 92.024 1291 90.864 1291 93.56 1290 88.848 1291 91.888 1291 89.328 1291 91.92 1290 90.112 1291 94.976 1291 92.384 1291 88.688 1291 88.112 1291 90.832 1291 93.176 1291 91.216 1291 90.752 1291 89.344 1291 88.528 1291 88.768 1291 94.784 1291 90.352 1291 92.336 1291 90.32 1290 90.016 1291 90.672 1291 89.36 1290 88.304 1291 89.392 1291 94.808 1291 88.464 1291 88.56 1290 88.032 1291 91.776 1291 91.424 1291 91.552 1291 90.304 1291 94.304 1291 91.008 1291 93.416 1291 94.592 1291 90.512 1291 91.952 1291 88.0 1290 92.096 1291 88.32 1290 88.944 1291 91.68 1290 88.416 1291 89.104 1291 88.8 1290 91.712 1291 88.496 1291 90.288 1291 93.008 1291 90.272 1291 90.176 1291 92.36 1290 90.032 1291 92.048 1291 94.544 1291 91.136 1291 88.992 1291 93.704 1291 90.576 1291 89.2 1290 88.336 1291 90.16 1290 90.256 1291 88.976 1291 89.072 1291 93.488 1291 91.296 1291 89.008 1291 90.608 1291 94.664 1291 90.384 1291 90.08 1290 94.736 1291 92.744 1291 91.088 1291 94.952 1291 90.128 1291 94.136 1291 90.544 1291 88.128 1291 90.768 1291 93.896 1291 91.344 1291 91.792 1291 92.936 1291 94.352 1291 94.424 1291 88.192 1291 88.72 1290 88.752 1291 93.824 1291 91.936 1291 88.256 1291 90.816 1291 94.4 1290 90.224 1291 90.8 1290 90.992 1291 92.144 1291 91.808 1291 92.24 1290 94.256 1291 94.088 1291 90.704 1291 91.824 1291 88.384 1291 93.68 1290 90.464 1291 92.72 1290 91.632 1291 91.36 1290 92.864 1291 93.2 1290 90.416 1291 89.248 1291 94.76 1290 93.32 1290 93.536 1291 90.064 1291 94.376 1291 88.272 1291 94.448 1291 89.376 1291 88.016 1291 94.688 1291 90.784 1291 93.296 1291 89.136 1291 93.848 1291 94.568 1291 91.52 1290 94.616 1291 93.08 1290 88.144 1291 92.168 1291 94.64 1290 94.184 1291 91.072 1291 90.624 1291 93.776 1291 93.512 1291 91.856 1291 88.64 1290 92.648 1291 88.512 1291 91.104 1291 90.848 1291 91.248 1291 91.264 1291 90.208 1291 90.912 1291 88.672 1291 88.88 1290 94.904 1291 92.84 1290 91.376 1291 89.424 1291 89.184 1291 91.456 1291 92.888 1291 90.976 1291 91.04 1290 93.272 1291 89.472 1291 93.608 1291 88.576 1291 90.688 1291 90.4 1290 91.504 1291 89.056 1291 90.448 1291 88.48 1290 91.408 1291 88.096 1291 88.176 1291 88.368 1291 91.744 1291 90.64 1290 94.832 1291 90.592 1291 93.656 1291 90.48 1290 91.12 1290 93.992 1291 92.504 1291 92.816 1291 91.152 1291 88.448 1291 90.496 1291 89.312 1291 93.224 1291 88.704 1291 90.736 1291 93.968 1291 92.216 1291 93.92 1290 90.144 1291 92.312 1291 91.056 1291 93.728 1291 89.168 1291 89.28 1290 88.048 1291 91.984 1291 92.456 1291 91.392 1291 92.192 1291 92.576 1291 88.96 1290 91.584 1291 94.928 1291 94.52 1290 92.96 1290 91.568 1291 90.656 1291 92.528 1291 88.816 1291 90.96 1290 88.624 1291 94.04 1290 88.24 1290 90.432 1291 90.048 1291 94.496 1291 94.88 1290 93.872 1291 94.472 1291 92.6 1290 88.928 1291 93.464 1291 90.88 1290 93.368 1291 93.752 1291 89.408 1291 92.624 1291 90.192 1291 92.072 1291 91.44 1290 90.928 1291 90.896 1291 94.232 1291 94.28 1290 92.792 1291 92.288 1291 92.432 1291 91.648 1291 93.44 1290 90.368 1291 93.944 1291 90.528 1291 89.04 1290 91.472 1291 89.12 1290 88.544 1291 92.696 1291 92.552 1291 91.536 1291 93.056 1291 93.032 1291 90.72 1290 93.584 1291 92.984 1291 92.768 1291 88.592 1291 92.12 1290 88.784 1291 91.6 1290 91.024 1291 88.16 1290 94.208 1291 92.264 1291 91.76 1290 93.104 1291 94.112 1291 88.832 1291 95.0 1290 90.24 1290 91.968 1291 89.264 1291 88.912 1291 88.208 1291 94.016 1291 91.168 1291 92.48 1290 92.0 1290 94.856 1291 88.896 1291 94.16 1290 89.152 1291 88.432 1291 91.84 1290 88.656 1291 93.128 1291 90.096 1291 93.8 1290 88.4 1290 91.872 1291 91.616 1291 93.632 1291 91.232 1291 91.2 1290 91.328 1291 94.064 1291 89.456 1291 88.08 1290 90.56 1290 91.312 1291 92.912 1291 91.664 1291 88.288 1291 88.608 1291 91.488 1291 88.352 1291 93.392 1291 91.28 1290 89.232 1291 89.44 1290 91.728 1291 90.944 1291 93.344 1291 89.296 1291 88.736 1291 93.248 1291 94.328 1291 93.152 1291 91.184 1291 94.712 1291 89.088 1291 88.864 1291 89.024 1291 90.0 

Output can only be rendered in Databricks

In [0]:
broj_temp_visok_df = (all_data_df.select('stres', 'temperatura_tela')
               .where(col('stres') == 3))
broj_temp_visok_df = (broj_temp_visok_df.groupBy(col('temperatura_tela')
                                     .alias('temperatura_tela')).count())

print('Broj osoba koda koje imaju visok stres: ', broj_temp_visok_df.count())
display(broj_temp_visok_df)

Broj osoba koda koje imaju visok stres:  94


temperatura_tela count 90.336 1291 90.864 1291 90.112 1291 90.832 1291 91.216 1291 90.752 1291 90.352 1291 90.016 1291 90.672 1291 90.32 1290 91.424 1291 90.304 1291 91.008 1291 90.512 1291 90.288 1291 90.272 1291 90.176 1291 90.032 1291 91.136 1291 90.576 1291 90.16 1290 90.256 1291 91.296 1291 90.608 1291 90.384 1291 90.08 1290 90.128 1291 91.088 1291 90.544 1291 90.768 1291 91.344 1291 90.816 1291 90.224 1291 90.992 1291 90.8 1290 90.704 1291 90.464 1291 91.36 1290 90.416 1291 90.064 1291 90.784 1291 91.072 1291 90.624 1291 91.104 1291 91.264 1291 91.248 1291 90.848 1291 90.208 1291 90.912 1291 91.376 1291 91.456 1291 91.04 1290 90.976 1291 90.448 1291 90.688 1291 90.4 1290 91.408 1291 90.64 1290 90.592 1291 90.48 1290 91.12 1290 91.152 1291 90.496 1291 90.144 1291 90.736 1291 91.056 1291 91.392 1291 90.656 1291 90.96 1290 90.432 1291 90.048 1291 90.88 1290 90.192 1291 90.928 1291 91.44 1290 90.896 1291 90.368 1291 90.528 1291 91.472 1291 90.72 1290 91.024 1291 90.24 1290 91.168 1291 90.096 1291 91.2 1290 91.232 1291 91.328 1291 91.312 1291 90.56 1290 91.488 1291 91.28 1290 90.944 1291 91.184 1291 90.0 1290

Output can only be rendered in Databricks

In [0]:
broj_kis_visok_df = (all_data_df.select('stres', 'kiseonik_u_krvi')
               .where(col('stres') == 3))
broj_kis_visok_df = (broj_kis_visok_df.groupBy(col('kiseonik_u_krvi')
                                     .alias('kiseonik_u_krvi')).count())

print('Broj osoba koda koje imaju visok stres: ', broj_kis_visok_df.count())
display(broj_kis_visok_df)

Broj osoba koda koje imaju visok stres:  94


kiseonik_u_krvi count 89.488 1291 88.848 1291 89.328 1291 88.688 1291 88.112 1291 89.344 1291 88.528 1291 88.768 1291 89.36 1290 88.304 1291 89.392 1291 88.464 1291 88.56 1290 88.032 1291 88.0 1290 88.32 1290 88.944 1291 88.416 1291 89.104 1291 88.8 1290 88.496 1291 88.992 1291 89.2 1290 88.336 1291 88.976 1291 89.072 1291 89.008 1291 88.128 1291 88.192 1291 88.72 1290 88.752 1291 88.256 1291 88.384 1291 89.248 1291 88.272 1291 89.376 1291 88.016 1291 89.136 1291 88.144 1291 88.64 1290 88.512 1291 88.672 1291 88.88 1290 89.424 1291 89.184 1291 89.472 1291 88.576 1291 89.056 1291 88.48 1290 88.096 1291 88.176 1291 88.368 1291 88.448 1291 89.312 1291 88.704 1291 89.168 1291 89.28 1290 88.048 1291 88.96 1290 88.816 1291 88.624 1291 88.24 1290 88.928 1291 89.408 1291 89.04 1290 89.12 1290 88.544 1291 88.592 1291 88.784 1291 88.16 1290 88.832 1291 89.264 1291 88.912 1291 88.208 1291 88.896 1291 89.152 1291 88.432 1291 88.656 1291 88.4 1290 89.456 1291 88.08 1290 88.288 1291 88.608 1291 89.232 1291 88.352 1291 89.44 1290 89.296 1291 88.736 1291 89.088 1291 88.864 1291 89.024 1291 88.064 1291 88.224 1291 89.216 1291

Output can only be rendered in Databricks

In [0]:
broj_temp_df.select(avg('temperatura_tela').alias('prosek temperature')).show()
broj_temp_visok_df.select(avg('temperatura_tela').alias('prosek temperature - visok stres')).show()


+------------------+
|prosek temperature|
+------------------+
| 93.11285797101449|
+------------------+

+--------------------------------+
|prosek temperature - visok stres|
+--------------------------------+
|                          90.744|
+--------------------------------+



In [0]:
broj_kis_df.select(avg('kiseonik_u_krvi').alias('prosek kiseonika')).show()
broj_kis_visok_df.select(avg('kiseonik_u_krvi').alias('prosek kiseonika - visok stres')).show()

+-----------------+
| prosek kiseonika|
+-----------------+
|91.29546666666661|
+-----------------+

+------------------------------+
|prosek kiseonika - visok stres|
+------------------------------+
|             88.74399999999996|
+------------------------------+



Na osnovu analize podataka i izdvajanju uzoraka dolazimo do sledećeg:  
visok stres je uticao i na temperaturu i na nivo kiseonika u krvi i to na način takav da prosečan nivo kiseonika je veći od prosečnog nivoa kiseonika tokom visokog stresa.  
Međutim, kod temperature je identična situacija kao kod nivoa kiseonika u krvi, što se ne slaže sa početnom hipotezom i što nas dovodi do zaključka da je pretpostavka bila <b>netačna</b>.

##Primena algoritma mašinskog učenja

U nastavku analizu podataka ćemo odraditi primenom algoritma mašinskog učenja. Biće korišćen višeslojni perceptron (MLP) koji spada u kategoriju algoritama sa nadgledanim učenjem (supervised learning). Najčešće se koristi za regresiju i klasifikaciju podataka.  

Kroz više slojeva ćemo prikazati klasifikaciju podataka. Zamisao je da pored transformisanih podataka koji se učitavaju iz DBFS-a, postoje tri sloja koja će po nivoima imati različitu topologiju. Na samom kraju ćemo imati softmax funkciju koja će u fazi testiranja da izvuče najbolje rezultate koje ćemo dalje proslediti na izlaz.

Pre početka rada predstavićemo način obrade podataka kroz tri faze: treniranje, tesiranje i validacije.  
Za treniranje ćemo izdvojiti 80% uzoraka, dok ćemo testiranje i validaciju objediniti sa 20% uzoraka.  
Kod ciklusa treniranja kao funkciju aktivacije ćemo koristiti tanh (hiperbolički tangens). U fazi testiranja kao što sam pomenuo prmenićemo Softmax funkciju, koju ne moramo implicinto da navodimo, jer je u sparku ona određenim importima uključena (MultilayerPerceptronClassificationModel).

In [0]:
!pip install tensorflow

     |▏                

*** WARNING: max output size exceeded, skipping output. ***

     |████████████████████████████████| 4.9 MB 25.7 MB/s 
     |████████████████████████████████| 781 kB 45.8 MB/s 
     |████████████████████████████████| 232 kB 43.8 MB/s 
     |████████████████████████████████| 177 kB 51.3 MB/s 
     |████████████████████████████████| 155 kB 48.5 MB/s 
     |████████████████████████████████| 77 kB 7.3 MB/s 
     |████████████████████████████████| 151 kB 54.4 MB/s 
  Attempting uninstall: MarkupSafe
    Found existing installation: MarkupSafe 2.0.1
    Not uninstalling markupsafe at /databricks/python3/lib/python3.9/site-packages, outside environment /local_disk0/.ephemeral_nfs/envs/pythonEnv-1a5ef4ca-7905-4cf9-826f-d75f82f6d10a
    Can't uninstall 'MarkupSafe'. No files were found to uninstall.
  Attempting uninstall: protobuf
    Found existing installation: protobuf 4.21.5
    Not uninstalling protobuf at /databricks/python3/lib/python3.9/site-packages, outside e

U notebook je uvežena biblioteka TensorFlow, zbog potreba u fazi treniranja.

In [0]:
import tensorflow as tf
from pyspark.sql import SparkSession
spark = SparkSession.builder.appName("stresAnalizaPodataka").config("spark.jars.packages", "org.tensorflow:spark-tensorflow-connector_2.12:1.15.0").getOrCreate()

In [0]:
from pyspark.ml.feature import VectorAssembler 
from pyspark.ml.feature import StringIndexer
from pyspark.ml.classification import LogisticRegressionModel
from pyspark.ml.evaluation import MulticlassClassificationEvaluator
from pyspark.ml.classification import MultilayerPerceptronClassifier
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.optimizers import SGD

In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from pyspark.sql import Row
from pyspark.sql.functions import col

spark = SparkSession.builder.appName("stresAnalizaPodataka").getOrCreate()

df = spark.read.format("csv").option("header", "true").load(all_data_df_path)

Učitavanje podataka i kreiranje df. Neophodno je bilo odraditi cast u narednom koraku.

In [0]:
df = df.withColumn('hrkanje', col('hrkanje').cast('double'))
df = df.withColumn('brzina_disanja', col('brzina_disanja').cast('double'))
df = df.withColumn('temperatura_tela', col('temperatura_tela').cast('double'))
df = df.withColumn('kiseonik_u_krvi', col('kiseonik_u_krvi').cast('double'))
df = df.withColumn('dužina_sna', col('dužina_sna').cast('double'))
df = df.withColumn('puls', col('puls').cast('double'))
df = df.withColumn('stres', col('stres').cast('int'))
df = df.withColumn('vlažnost', col('vlažnost').cast('double'))
df = df.withColumn('broj_koraka', col('broj_koraka').cast('int'))
df = df.withColumn('nivo_stresa', col('nivo_stresa').cast('int'))

Radi dobijanja što boljih rezultata prilikom ove analize odlučilo sam da odbacim sledeće dve kolone.

In [0]:
df = df.drop('vlažnost', 'nivo_stresa')

display(df)

hrkanje brzina_disanja temperatura_tela kiseonik_u_krvi dužina_sna puls stres broj_koraka 60.0 20.0 96.0 95.0 7.0 60.0 1 51 85.76 23.536 90.768 88.768 0.768 68.84 3 51 56.88 19.376 95.376 94.064 6.376 58.44 1 51 55.52 19.104 95.104 93.656 6.104 57.76 1 51 73.44 21.344 93.344 91.344 4.016 63.36 2 51 59.28 19.856 95.856 94.784 6.856 59.64 1 51 87.8 24.08 91.04 89.04 1.04 70.2 3 51 52.32 18.464 94.464 92.696 5.464 56.16 1 51 52.64 18.528 94.528 92.792 5.528 56.32 1 51 86.24 23.664 90.832 88.832 0.832 69.16 3 51 81.56 22.416 90.208 88.208 0.208 66.04 3 51 63.68 20.368 92.368 90.368 2.552 60.92 2 51 77.6 21.76 93.76 91.76 4.64 64.4 2 51 77.28 21.728 93.728 91.728 4.592 64.32 2 51 69.76 20.976 92.976 90.976 3.464 62.44 2 51 88.04 24.144 91.072 89.072 1.072 70.36 3 51 89.96 24.656 91.328 89.328 1.328 71.64 3 51 53.68 18.736 94.736 93.104 5.736 56.84 1 51 78.56 21.856 93.856 91.856 4.784 64.64 2 51 50.96 18.192 94.192 92.288 5.192 55.48 1 51 62.08 20.208 92.208 90.208 2.312 60.52 2 51 60.0 20.0 92.0 90.0 2.0 60.0 2 51 86.72 23.792 90.896 88.896 0.896 69.48 3 51 60.96 20.096 92.096 90.096 2.144 60.24 2 51 65.6 20.56 92.56 90.56 2.84 61.4 2 51 75.36 21.536 93.536 91.536 4.304 63.84 2 51 73.28 21.328 93.328 91.328 3.992 63.32 2 51 62.88 20.288 92.288 90.288 2.432 60.72 2 51 56.72 19.344 95.344 94.016 6.344 58.36 1 51 53.6 18.72 94.72 93.08 5.72 56.8 1 51 56.08 19.216 95.216 93.824 6.216 58.04 1 51 75.68 21.568 93.568 91.568 4.352 63.92 2 51 87.56 24.016 91.008 89.008 1.008 70.04 3 51 58.48 19.696 95.696 94.544 6.696 59.24 1 51 81.08 22.288 90.144 88.144 0.144 65.72 3 51 61.6 20.16 92.16 90.16 2.24 60.4 2 51 50.24 18.048 94.048 92.072 5.048 55.12 1 51 60.48 20.048 92.048 90.048 2.072 60.12 2 51 50.48 18.096 94.096 92.144 5.096 55.24 1 51 57.68 19.536 95.536 94.304 6.536 58.84 1 51 56.48 19.296 95.296 93.944 6.296 58.24 1 51 90.08 24.688 91.344 89.344 1.344 71.72 3 51 50.64 18.128 94.128 92.192 5.128 55.32 1 51 50.56 18.112 94.112 92.168 5.112 55.28 1 51 53.92 18.784 94.784 93.176 5.784 56.96 1 51 68.32 20.832 92.832 90.832 3.248 62.08 2 51 54.88 18.976 94.976 93.464 5.976 57.44 1 51 52.4 18.48 94.48 92.72 5.48 56.2 1 51 51.6 18.32 94.32 92.48 5.32 55.8 1 51 65.12 20.512 92.512 90.512 2.768 61.28 2 51 86.6 23.76 90.88 88.88 0.88 69.4 3 51 68.0 20.8 92.8 90.8 3.2 62.0 2 51 63.2 20.32 92.32 90.32 2.48 60.8 2 51 75.2 21.52 93.52 91.52 4.28 63.8 2 51 76.96 21.696 93.696 91.696 4.544 64.24 2 51 80.0 22.0 90.0 88.0 0.0 65.0 3 51 53.52 18.704 94.704 93.056 5.704 56.76 1 51 61.44 20.144 92.144 90.144 2.216 60.36 2 51 62.24 20.224 92.224 90.224 2.336 60.56 2 51 79.2 21.92 93.92 91.92 4.88 64.8 2 51 54.4 18.88 94.88 93.32 5.88 57.2 1 51 68.96 20.896 92.896 90.896 3.344 62.24 2 51 90.56 24.816 91.408 89.408 1.408 72.04 3 51 66.88 20.688 92.688 90.688 3.032 61.72 2 51 68.64 20.864 92.864 90.864 3.296 62.16 2 51 76.0 21.6 93.6 91.6 4.4 64.0 2 51 54.96 18.992 94.992 93.488 5.992 57.48 1 51 55.28 19.056 95.056 93.584 6.056 57.64 1 51 90.32 24.752 91.376 89.376 1.376 71.88 3 51 58.08 19.616 95.616 94.424 6.616 59.04 1 51 83.6 22.96 90.48 88.48 0.48 67.4 3 51 65.76 20.576 92.576 90.576 2.864 61.44 2 51 57.2 19.44 95.44 94.16 6.44 58.6 1 51 68.8 20.88 92.88 90.88 3.32 62.2 2 51 59.68 19.936 95.936 94.904 6.936 59.84 1 51 78.08 21.808 93.808 91.808 4.712 64.52 2 51 71.36 21.136 93.136 91.136 3.704 62.84 2 51 81.44 22.384 90.192 88.192 0.192 65.96 3 51 78.4 21.84 93.84 91.84 4.76 64.6 2 51 68.48 20.848 92.848 90.848 3.272 62.12 2 51 50.32 18.064 94.064 92.096 5.064 55.16 1 51 88.52 24.272 91.136 89.136 1.136 70.68 3 51 64.64 20.464 92.464 90.464 2.696 61.16 2 51 74.56 21.456 93.456 91.456 4.184 63.64 2 51 70.56 21.056 93.056 91.056 3.584 62.64 2 51 55.76 19.152 95.152 93.728 6.152 57.88 1 51 66.08 20.608 92.608 90.608 2.912 61.52 2 51 76.8 21.68 93.68 91.68 4.52 64.2 2 51 59.6 19.92 95.92 94.88 6.92 59.8 1 51 74.72 21.472 93.472 91.472 4.208 63.68 2 51 71.2 21.12 93.12 91.12 3.68 62.8 2 51 58.16 19.632 95.632 94.448 6.632 59.08 1 51 53.44 18.688 94.688 93.032

In [0]:
#klasa koja nasleđuje tf.keras.Model
#potrebna nam je da bi se model mogao koristiti sa sparkom
class SparkModelWrapper(tf.keras.Model):
    def __init__(self, model):
        super(SparkModelWrapper, self).__init__()
        self.model = model

    def call(self, inputs):
        return self.model(inputs)
    
#kreiranje keras modela
keras_model = Sequential()
keras_model.add(Dense(64, activation='tanh', input_shape=(8,)))
keras_model.add(Dense(32, activation='tanh'))
keras_model.add(Dense(16, activation='tanh'))
keras_model.add(Dense(5, activation='tanh'))


In [0]:
#SparkModelWrapper služi za podešavanje keras modela
spark_model = SparkModelWrapper(keras_model)

#kompajliranje modela
sgd = SGD(learning_rate=0.001)
spark_model.compile(optimizer=sgd, loss='sparse_categorical_crossentropy', metrics=['accuracy'])


In [0]:
# kreiranje asembler vektora za spajanje kolona u jednu features kolonu
assembler = VectorAssembler(inputCols=["hrkanje", "brzina_disanja", "temperatura_tela", "kiseonik_u_krvi", "dužina_sna", "puls", "stres", "broj_koraka"], outputCol="features")

#dodajemo novu kolonu label koju smo izabrali kao ciljnu varijablu
#ona predstavlja nivo stresa čiji je sadržaj prikazan na pokačtku ovog notebook-a
df = df.withColumn("label", col("stres"))


In [0]:
#transformišemo izabrani data frame
transf = assembler.transform(df)

In [0]:
#ovim delimo set na podatke za treniranje koji čine 80% i podatke za testiranje i validaciju koji čine 20%
#(trainingData, testData) = transf.randomSplit([0.8, 0.2])
#trainingData.cache()

In [0]:
#treniranje modela
mlp = mlp = MultilayerPerceptronClassifier(featuresCol="features", labelCol="label", predictionCol="prediction", maxIter=10, blockSize=800, layers=[8, 64, 32, 16, 5], stepSize=0.001)
model = mlp.fit(assembler.transform(df))

In [0]:
(trainingData, testData) = transf.randomSplit([0.8, 0.2])
trainingData.cache()

#testiranje modela
predictions = model.transform(testData)

In [0]:
evaluator = MulticlassClassificationEvaluator(labelCol="label", predictionCol="prediction", metricName="accuracy")
accuracy = evaluator.evaluate(predictions)

print("Accuracy:", accuracy)

Accuracy: 0.7112620793255756


Na samom kraju dobijamo tačnost našeg algoritma koja iznosi približno 72%.  
To nas dovodi do zaključka da smo postigli veoma visoku tačnost u odnosu na očekivanu koju smo postavili na početku analize, a ta vrednost je bila 80%.  
A samim tim možemo reći da kao rezultat celokupne analize svi faktori imaju udela u razvoju visokog nivoa stresa kod čoveka.  
Takođe, imam i sliku višeslojnog perceptrona koja nažalost nije mogla da stane u ovaj markdown, ali biće poslata uz sam rad.

<img src="https://drive.google.com/file/d/1GElHDM121XgunLusyR7_HsYkXPY7Be5s/view?usp=share_link" alt="Višeslojni perceptron" />

##Structured Streaming

In [0]:
%fs ls dbfs:/streaming/df.csv

path name size modificationTime dbfs:/streaming/df.csv/_SUCCESS _SUCCESS 0 1676291040000 dbfs:/streaming/df.csv/_committed_3341368505843524836 _committed_3341368505843524836 113 1676291039000 dbfs:/streaming/df.csv/_started_3341368505843524836 _started_3341368505843524836 0 1676291036000 dbfs:/streaming/df.csv/part-00000-tid-3341368505843524836-1a17a0b3-ef84-4e77-bc79-aa901c29d024-538-1-c000.csv part-00000-tid-3341368505843524836-1a17a0b3-ef84-4e77-bc79-aa901c29d024-538-1-c000.csv 22843777 1676291038000

In [0]:
all_str_path = 'dbfs:/streaming/'

schema = StructType([
     StructField("stres", IntegerType(), True),
     StructField('timestamp', StringType(), True),
    # StructField('count', LongType(), True)
])

In [0]:
from pyspark.sql.functions import *
str_df = (spark
            .read
            .schema(schema)   
            .json(all_str_path))

#broj_str = (str_df.groupBy('stres').count())
#broj_str = (str_df.groupBy(str_df.stres, window(str_df.timestamp, "1 hour")).count())
#broj_str.isStreaming
display(str_df)

DataFrame[stres: int, timestamp: string]

In [0]:
from pyspark.sql.functions import *    

count_DF = (str_df.groupBy(str_df.stres, window(str_df.timestamp, "1 hour")).count())
count_DF.cache()

display(count_DF)

DataFrame[stres: int, window: struct<start:timestamp,end:timestamp>, count: bigint]

In [0]:
count_DF.createOrReplaceTempView("static_counts")

In [0]:
%sql
select stres, sum(count) as total_count from static_counts group by stres

DataFrame[stres: int, total_count: bigint]

In [0]:
# keep the size of shuffles small
spark.conf.set("spark.sql.shuffle.partitions", "2")  

#query = (str_df
#    .writeStream
#   .format("memory")        
#    .queryName("kiseonik_u_krvi")    
#    .outputMode("append")  
#    .start())

In [0]:
from time import sleep
sleep(5)

In [0]:
%sql select stres, sum(count) from nivo_stresa group by stres

DataFrame[stres: int, sum(count): bigint]

In [0]:
query.stop()